pkg/metrics/exporters/stackdriver/stackdriver.go

```diff
@@ -31,26 +36,22 @@ func Start(ctx context.Context) error {
 		return err
 	}
 
-	exporter, err := stackdriver.NewExporter(stackdriver.Options{
-		MetricPrefix:      metricPrefix,
-		MonitoredResource: monitoredresource.Autodetect(),
-	})
+	e, err := stackdriver.New(stackdriver.WithProjectID(metricPrefix))
 	if err != nil {
 		if *ignoreMissingCreds {
 			log.Error(err, "Error initializing stackdriver exporter, not exporting stackdriver metrics")
 			return nil
 		}
 		return err
 	}
-
-	if err := exporter.StartMetricsExporter(); err != nil {
-		if *ignoreMissingCreds {
-			log.Error(err, "Error starting stackdriver exporter, not exporting stackdriver metrics")
-			return nil
-		}
-		return err
-	}
-	defer exporter.StopMetricsExporter()
+	reader := metric.NewPeriodicReader(e, metric.WithInterval(time.Duration(*metricInterval)*time.Second))
+	meterProvider := metric.NewMeterProvider(
+		metric.WithReader(reader),
+		metric.WithView(view.Views()...),
```

Comment:
Looking at the metrics being exported, there's another gap because we don't set the `resource_type` now. Previously, it was `k8s_container` and now it defaults to `generic_node`.  I did some digging into how we can bridge the gap.

---

We need to specify the [`Detector`s  on the resource](https://opentelemetry.io/docs/instrumentation/go/resources/) here that bridge the gap between what we got from [Autodetect](https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/blob/v0.13.14/monitoredresource/monitored_resources.go#L39) :

```
// Autodetect auto detects monitored resources based on
// the environment where the application is running.
// It supports detection of following resource types
// 1. gke_container:
// 2. gce_instance:
// 3. aws_ec2_instance:
```

Luckily, we have those detectors already in the contrib:

* https://github.com/open-telemetry/opentelemetry-go-contrib/tree/main/detectors/gcp
* https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/detectors/aws/README.md 

Also, apart from the detectors, the GCP readme calls out:

> When using this detector, you should use this in your Pod Spec to set these using OTEL_RESOURCE_ATTRIBUTES:

So we will need to change the manifest too to include that as an env var.

Reply:
Thanks for looking into this @acpana, I added the code for detector and addressed all your feedback as well.